pandas/tests/generic/test_generic.py

```diff
@@ -18,124 +18,120 @@
 # Generic types test cases
 
 
-class Generic:
-    @property
-    def _ndim(self):
-        return self._typ._AXIS_LEN
-
-    def _axes(self):
-        """return the axes for my object typ"""
-        return self._typ._AXIS_ORDERS
-
-    def _construct(self, shape, value=None, dtype=None, **kwargs):
-        """
-        construct an object for the given shape
-        if value is specified use that if its a scalar
-        if value is an array, repeat it as needed
-        """
-        if isinstance(shape, int):
-            shape = tuple([shape] * self._ndim)
-        if value is not None:
-            if is_scalar(value):
-                if value == "empty":
-                    arr = None
-                    dtype = np.float64
-
-                    # remove the info axis
-                    kwargs.pop(self._typ._info_axis_name, None)
-                else:
-                    arr = np.empty(shape, dtype=dtype)
-                    arr.fill(value)
+def check_metadata(x, y=None):
```

Comment:
can you move this to _testing? and call it assert_metadata or similar (doc-string & typing if you can)